linker-interface/shared/src/test/scala/org/scalajs/linker/interface/LegacyV1_2AdapterTest.scala

```diff
@@ -0,0 +1,189 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.interface
+
+import scala.collection.mutable
+import scala.concurrent._
+
+import java.net.URI
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets.UTF_8
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalajs.junit.async._
+
+import org.scalajs.linker.interface.unstable._
+
+@deprecated("Mark deprecated to silence warnings", "never/always")
+class LegacyV1_2AdapterTest {
+  import LegacyV1_2AdapterTest._
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  private val dummyReport = new ReportImpl(List(
+      new ReportImpl.ModuleImpl(
+          moduleID = "dummy",
+          jsFileName = "main.js",
+          sourceMapName = Some("main.js.map"),
+          moduleKind = ModuleKind.NoModule
+      )
+  ))
+
+  @Test
+  def testReplaceLinks(): AsyncResult = await {
+    val writeOut = new WriteOnlyOutputDirectory()
+
+    val legacyOutput = LinkerOutput(new OutputFileImpl("js", writeOut))
+      .withSourceMap(new OutputFileImpl("sm", writeOut))
+      .withSourceMapURI(new URI("http://example.org/my-source-map-uri"))
+      .withJSFileURI(new URI("http://example.org/my-js-file-uri"))
+    
+    val readOut = new ReadOnlyOutputDirectory(
+        "main.js" -> raw"""
+        |console.log("hello");
+        |//# sourceMappingURL=main.js.map
+        |// some other comment
+        |""".stripMargin,
+        "main.js.map" -> raw"""{
+        |  "file": "main.js",
+        |  "other key": 1
+        |}""".stripMargin)
+
+    for {
+      _ <- LegacyV1_2Adapter.convert(dummyReport, readOut, legacyOutput)
+    } yield {
+      assertEquals(writeOut.content.size, 2)
+
+      assertEquals(raw"""
+          |console.log("hello");
+          |//# sourceMappingURL=http://example.org/my-source-map-uri
+          |// some other comment
+          |""".stripMargin, writeOut.content("js"))
+      assertEquals(raw"""{"file": "http://example.org/my-js-file-uri",
+          |  "other key": 1
+          |}""".stripMargin, writeOut.content("sm"))
+    }
+  }
+
+  @Test
+  def testAddLinks(): AsyncResult = await {
+    val writeOut = new WriteOnlyOutputDirectory()
+
+    val legacyOutput = LinkerOutput(new OutputFileImpl("js", writeOut))
+      .withSourceMap(new OutputFileImpl("sm", writeOut))
+      .withSourceMapURI(new URI("http://example.org/my-source-map-uri"))
+      .withJSFileURI(new URI("http://example.org/my-js-file-uri"))
+    
+    val readOut = new ReadOnlyOutputDirectory(
+        "main.js" -> raw"""
+        |console.log("hello");
+        |""".stripMargin,
+        "main.js.map" -> raw"""{
+        |  "other key": 1
+        |}""".stripMargin)
```

Comment:
The indentation in these examples is confusing. It's not clear where elements start and where they end. Perhaps indent it as follows:
```suggestion
        "main.js" -> raw"""
          |console.log("hello");
          |""".stripMargin,
        "main.js.map" -> raw"""{
          |  "other key": 1
          |}""".stripMargin)
```
?